src/bin/pg_autoctl/pgsql.c

```diff
@@ -1178,8 +1178,11 @@ pgsql_has_replica(PGSQL *pgsql, char *userName, bool *hasReplica)
 	context.resultType = PGSQL_RESULT_BOOL;
 	context.parsedOk = false;
 
-	pgsql_execute_with_params(pgsql, sql, paramCount, paramTypes, paramValues,
-							  &context, &parseSingleValueResult);
+	if (!pgsql_execute_with_params(pgsql, sql, paramCount, paramTypes, paramValues,
+								   &context, &parseSingleValueResult))
+	{
+		return false;
```

Comment:
What about logs? at least a comment saying that `/* errors have already been logged */` after verifying that this is indeed the case?